lib/s3.js

```diff
@@ -17,32 +18,26 @@ const S3 = {};
  * Zips a local directory and uploads it to S3
  * @param {*} localDir
  */
-S3.uploadZip = async function (localDir) {
+S3.uploadZip = async function (pathToZip) {
   if (!process.env.AWS_S3_BUCKET) {
     throw new Error('To use S3, you must specify AWS_S3_BUCKET in environment variables');
   }
-  if (!fs.exists(localDir)) {
-    throw new Error(`Could not find directory ${localDir}`);
-  }
-
-  // Attempt to zip the directory
-  let base64Zip;
-  try {
-    base64Zip = await zip.toInMemoryZip(localDir);
-  } catch (err) {
-    throw new Error(`Could not zip directory ${localDir}: ${err}`);
+  if (!fs.exists(pathToZip)) {
+    throw new Error(`Could not find directory ${pathToZip}`);
   }
 
   // Call s3.upload and return it as a promise
   // (NOTE: B.promisify doesn't work for s3.upload, it causes s3/managed_upload.js to throw an error `self.service.constructor.__super__ is not a constructor`)
-  return await new B((resolve, reject) => {
-    s3.upload({
-      Bucket: process.env.AWS_S3_BUCKET,
-      Key: `${uuid()}.zip`, // Give it a random name
+  logger.debug(`Uploading ${pathToZip} to S3 ${Bucket}`);
+  return await new B(async (resolve, reject) => {
+    const uploader = s3.upload({
+      Bucket,
+      Key: `${uuid()}.zip`, // Name it with a GUID to avoid conflicts
       ACL: 'public-read',
       Expiration: 1, // Expires after one day
-      Body: base64Zip,
-    }, (err, res) => err ? reject(new Error(`Could not upload ${localDir} to S3. Check that you have set the S3 environment variables correctly: ${err}`)) : resolve(res));
+      Body: await fs.readFile(pathToZip),
```

Comment:
i think it'd be better to call `await fs.readFile(pathToZip)` outside this promise, so that this function doesn't need to be an `async` function. currently, if `fs.readFile(pathToZip)` throws an error, or indeed if anything else in the body of this function throws an error, it will become an unhandled rejection. that is because nothing is `await`ing this function, so errors are not bubbled up appropriately.

Reply:
Once that is shifted you should be able to use `B.promisify` on `s3.upload` and not need to manually do any of this Promise handling.